java/code/src/org/keyczar/KeyczarKey.java

```diff
@@ -84,6 +87,8 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Util.toInt(this.hash());
   }
+  
+  protected Stream cachedStream = null;
```

Comment:
And just to be clear, if the cache needs stay. The intent of this change was that it needs to be on the key instance instead of in a global key map global for the possibility of key collisions.